tests/queries/tests.py

```diff
@@ -2248,6 +2248,44 @@ def test_field_error_values_list(self):
         with self.assertRaisesMessage(FieldError, msg):
             Tag.objects.values_list('name__foo')
 
+    def test_named_values_list_flat(self):
+        msg = "'flat' and 'named' can't be passed together."
+        with self.assertRaisesMessage(TypeError, msg):
+            Number.objects.values_list('num', flat=True, named=True)
+
+    def test_named_values_list_bad_field_name(self):
+        msg = "Type names and field names must be valid identifiers: '1'"
+        with self.assertRaisesMessage(ValueError, msg):
+            Number.objects.extra(select={'1': 'num+1'}).values_list('1', named=True).first()
```

Comment:
Is it possible to trigger the error without using `QuerySet.extra()` considering that it might be deprecated in the future?

Reply:
It seems that `QuerySet.extra()` is the only way to make field name which is not valid identifier.